deepr/layers/core.py

```diff
@@ -160,3 +160,17 @@ def forward(self, tensors, mode: str = None):
             use_bias=self.use_bias,
             **self._kwargs
         )
+
+
+class Softmax(base.Layer):
+    """Apply softmax to the last dimension of tensor with filtering masked values"""
+
+    def __init__(self, n_in: int = 2, n_out=1, **kwargs):
+        super().__init__(n_in=n_in, n_out=n_out, **kwargs)
+
+    def forward(self, tensors, mode: str = None):
+        """Forward method of the layer"""
+        tensor, mask = tensors
+        tensor_exp = tf.exp(tensor - tf.reduce_max(tensor, axis=-1, keepdims=True))
```

Comment:
the reduce max is here for numerical stability right?
Also, shouldn't it be also multiplied by the mask here?